### PR TITLE
MAVLink improvement

### PR DESCRIPTION
**Pull request by shengwen1997**
*Three message parsers: generic, mission and parameter
*Support handling different transactions at the same time
*Broadcast task for sending quad info message only
*Receiver task for ground station message/transaction/command handling
*New timeout mechanism

And more...

If the Pull Request is ready to merge, please notify me to disable the mavlink DEBUG print setting.

### CODE REVIEW

**Comment by fboris** [comment]
Maybe we can change `communication.c` to more general name?
e.g. `mavlink_manager.c` ?

**Comment by shengwen1997** [comment]
How about "mavlink_task"?

**Comment by fboris** [comment]
It does not only contain task.

**Comment by fboris** [comment]
any proposal now?

**Comment by fboris** [comment]
Did you merge Ming's new commita?

**Comment by shengwen1997** [comment]
Could you give me the link?

File: program/mavlink/communication.h [hunk 1]
```diff
@@ -7,10 +7,34 @@
 #include "mavlink.h"
 #include "communication.h"
 
-void send_package(mavlink_message_t *msg);
-void clear_message_id(mavlink_message_t *message);
+#define BROADCAST_MESSAGE_CNT (sizeof(boradcast_message_list) / sizeof(boradcast_message_list[0]))
```
Comment (fboris) [inline program/mavlink/communication.h]: typo `boradcast_message_list`

**Comment by fboris** [comment]
Could you change target branch to 'dev/mavlink'? I don't want to merge it too soon. And Please change 'communication.c'c to reasonable one.

**Pull request closed without merging**